d2/src/main/java/com/linkedin/d2/balancer/D2ClientConfig.java

```diff
@@ -135,6 +137,7 @@ public class D2ClientConfig
   public String xdsChannelLoadBalancingPolicy = null;
   public Map<String, ?> xdsChannelLoadBalancingPolicyConfig = null;
   public boolean subscribeToUriGlobCollection = false;
+  public XdsServerMetricsProvider _xdsServerMetricsProvider = new NoOpXdsServerMetricsProvider();
```

Comment:
true. Just in case the constructor gets changed in the future by mistake, wanted to double make sure the no-op one is used. Can't be too careful.

Reply:
you mean D2ClientlConfig constructor or XdsClientImpl constructor?

For D2ClientlConfig constructor, if D2ClientBuilder's setXdsServerMetricsProvider is not called, the builder will just use the default _xdsServerMetricsProvider value in its _config, which is what's initialized here. And it will be passed to XdsClientImpl constructor, so this can prevent null being passed to XdsClientImpl constructor (if someone directly call XdsClientImpl constructor with null, this can't prevent null being used).

Reply:
Since it's open source code, someone outside LI can directly create XdsClientImpl without using D2ClientConfig for their own purposes.

For us, D2ClientConfig needs _xdsServerMetricsProvider for container to pass the actual INDIS implementation of the provider to restli code. We're using the full XdsClientImpl constructor to pass in the LI specific provider, but people outside LI can be totally ok not using it. 
It's a common pattern in many classes when involving passing LI specific impl to some public classes.